runtime/jcl/common/mgmtmempool.c

```diff
@@ -173,14 +178,21 @@ Java_com_ibm_java_lang_management_internal_MemoryPoolMXBeanImpl_getUsageImpl(JNI
 		UDATA idx = getIndexFromPoolID(mgmt, id);
 		J9MemoryPoolData *pool = &mgmt->memoryPools[idx];
 
-		javaVM->memoryManagerFunctions->j9gc_pools_memory(javaVM, (id&J9VM_MANAGEMENT_POOL_HEAP_ID_MASK), &totals[0], &frees[0], FALSE);
-		committed = totals[idx];
-		used = committed - frees[idx];
+		UDATA total = 0;
+		UDATA free = 0;
+		UDATA maximum = 0;
+
+		/* acquire vmAccess for retrieving memoryUsage for the pool in order to guarantee values of memoryUsage are consistent. */
+		javaVM->internalVMFunctions->internalAcquireVMAccess(vmThread);
```

Comment:
I think this should be the internalEnterVMFromJNIcall instead

Reply:
done